src/parser.ts

```diff
@@ -306,17 +308,44 @@ function parseSuperTypes(
   options: Options,
   processed: Processed,
   usedNames: UsedNames
-): TNamedInterface[] {
+): TSuperType[] {
   // Type assertion needed because of dereferencing step
   // TODO: Type it upstream
   const superTypes = schema.extends as SchemaSchema | SchemaSchema[] | undefined
   if (!superTypes) {
     return []
   }
   if (Array.isArray(superTypes)) {
-    return superTypes.map(_ => newNamedInterface(_, options, _, processed, usedNames))
+    return superTypes.map(_ => newSuperType(_, options, _, processed, usedNames))
+  }
+  return [newSuperType(superTypes, options, superTypes, processed, usedNames)]
+}
+
+function newSuperType(
+  schema: SchemaSchema,
+  options: Options,
+  rootSchema: JSONSchema,
+  processed: Processed,
+  usedNames: UsedNames
+): TSuperType {
+  if (schema.tsType) {
+    return newExtendedType(schema, usedNames)
+  }
+  return newNamedInterface(schema, options, rootSchema, processed, usedNames)
+}
+
+function newExtendedType(schema: SchemaSchema, usedNames: UsedNames): TCustomType {
+  if (schema.tsType) {
+    return {
+      comment: schema.description,
+      keyName: schema.title,
+      params: schema.tsType,
+      standaloneName: standaloneName(schema, schema.title, usedNames),
+      type: 'CUSTOM_TYPE'
+    }
+  } else {
+    throw Error(format('Customtype must have a tsType', schema))
```

Comment:
This method is only called in 1 place where `schema.tsType` is already checked so we should be good, this is mostly a guard against programming mistakes.